model_analyzer/config/input/objects/config_model_analysis_spec.py

```diff
@@ -80,11 +97,21 @@ def set_constraints(self, constraints):
         Parameters
         -------
         constraints : dict or None
-            A dictionary containing the parameters
+            A dictionary containing the constraints
         """
 
         self._constraints = constraints
 
+    def set_weightings(self, weightings):
```

Comment:
Do we need this function? I don't see it called, and I don't think we have similar functions for constraints or objectives?

Reply:
It is not called, but we do have set functions for objectives and constraints, so I thought it was good to add this. I thought it was good/clean for a class to have get/set methods for private variables, regardless of whether they are currently used.

Reply:
I definitely wouldn't call it clean to always have get/set. If the values should be set on construction and never modified then we don't want a set function.

I see now that there are set functions for constraints/objectives/model name. They are never called and should be ripped out. Those aren't things we ever expect to change.

Reply:
Removed